src/index.ts

```diff
@@ -78,36 +78,30 @@ export default async function GleeAppInitializer() {
   await generateDocs(parsedAsyncAPI, config, null)
 
   channelNames.forEach((channelName) => {
-    const channel = parsedAsyncAPI.channel(channelName)
-    if (channel.hasPublish()) {
-      const operationId = channel.publish().json('operationId')
+    const channel = parsedAsyncAPI.channels().get(channelName)
+    channel.operations().filterByReceive().forEach(operation => {
+      const operationId = operation.operationId()
+
       if (operationId) {
         const schema = {
-          oneOf: channel
-            .publish()
-            .messages()
-            .map((message) => message.payload().json()),
+          oneOf: operation.messages().filterByReceive().map(m => m.payload())
         } as any
         app.use(channelName, validate(schema), (event, next) => {
           triggerFunction({
             app,
             operationId,
-            message: event,
-          })
-            .then(next)
-            .catch(next)
+            message: event
+          }).then(next).catch(next)
         })
       }
-    }
-    if (channel.hasSubscribe()) {
+    })
+
+    channel.operations().filterBySend().forEach(operation => {
       const schema = {
-        oneOf: channel
-          .subscribe()
-          .messages()
-          .map((message) => message.payload().json()),
+        oneOf: operation.messages().filterBySend().map(m => m.payload())
```

Comment:
```suggestion
        oneOf: operation.messages().filterBySend().map(m => m.payload().json())
```
it doesn't work for me without the `.json()` in `dummy` example